docs/spec.md

```diff
@@ -44,6 +44,62 @@ If a method called with incorrect types the behaviour is unspecified.
 Also if a method accepts a function it must only apply the function in accordance with
 the type signature (i.e. provide the correct number of arguments of the correct types).
 
+
+## Parametricity
+
+All methods' implementations should only use type information about arguments that is known from the
+methods' type signatures. It's not allowed to inspect arguments or values that they produce
+or contain to get more information about their types. In other words methods
```

Comment:
I don't like that it says "... to get more information about their types". When we do something like `a === undefined` we don't really inspect to get more information about type but rather use `===` operation that works with any types, so it kinda technically legitimate.

Don't know how to say it better. Any suggestions?